sympy/functions/special/tests/test_hyper.py

```diff
@@ -343,3 +343,27 @@ def test_limits():
            O(k**6) # issue 6350
     assert limit(meijerg((), (), (1,), (0,), -x), x, 0) == \
             meijerg(((), ()), ((1,), (0,)), 0) # issue 6052
+
+def test_appellf1():
+    a, b1, b2, c, x, y = symbols('a b1 b2 c x y')
+    assert appellf1(a, b2, b1, c, y, x) == appellf1(a, b1, b2, c, x, y)
+    assert appellf1(a, b1, b1, c, y, x) == appellf1(a, b1, b1, c, x, y)
+
+def test_derivative_appellf1():
+    from sympy import diff
+    a, b1, b2, c, x, y, z = symbols('a b1 b2 c x y z')
+    assert diff(appellf1(a, b1, b2, c, x, y), x) == a*b1*appellf1(a + 1, b2, b1 + 1, c + 1, y, x)/c
+    assert diff(appellf1(a, b1, b2, c, x, y), y) == a*b2*appellf1(a + 1, b1, b2 + 1, c + 1, x, y)/c
+    assert diff(appellf1(a, b1, b2, c, x, y), z) == 0
+
+def test_appellf1_rewrite_factorial():
+    from sympy import factorial, Sum, oo
+    x, y = symbols('x y')
+    m, n = symbols('m n', integer=True)
+    assert appellf1(1 ,2 ,2, 1, 0.5, 0.6).rewrite(factorial) == \
+        Sum(0.5**m*0.6**n*factorial(m + 1)*factorial(n + 1)/(factorial(m)*\
+        factorial(n)), (m, 0, oo), (n, 0, oo))
```

Comment:
I was unable to get these expressions in Mathematica. Did you use Mathematica to verify these results?

Reply:
`RisingFactorial.rewrite(factorial)` gives wrong result in sympy. I think we don't need this. Once this issue (#14871) is fixed, it can be added.